### ISSUE TITLE
Embedded list query problem

### ISSUE THREAD

**Issue by MuratAkbaba006** [realm/realm-js#5865]
### How frequently does the bug occur?

Always

### Description

Hello, I want to query an embedded list from this schema
![Screen Shot 2023-06-02 at 16 26 08](https://github.com/realm/realm-js/assets/44754934/b5f9dde3-f7db-4fb6-9b2e-559370feb33a)

I am trying this query 
" locationType = 'city' and data.mvFdwLocationCode = 'tr-56' "

![Screen Shot 2023-06-02 at 16 35 20](https://github.com/realm/realm-js/assets/44754934/4765863d-1349-4d78-b78e-950a207f15c8)

this query returns all of the citites as you can see from the screen shot but I have only one city with this location code.
Am I doing something wrong or isn't is possible to query embedded list?. 

Also this is the city schema
![Screen Shot 2023-06-02 at 16 26 45](https://github.com/realm/realm-js/assets/44754934/973e5b16-ad37-476c-aa2e-a22341873b21)

Thank you.

### Can you reproduce the bug?

Always

### Reproduction Steps

`const queryString = `locationType = 'city' and data.mvFdwLocationParentCode == 'tr-63'`;
    const data = realm.objects('generalLocation').filtered(queryString);`

### Version

11.8.0

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

Mac Monterey 12.5.1 

### Build environment

Which debugger for React Native: ..
none

### Cocoapods version

cocoapods-1.11.3

**Comment by kneth** [comment]
Correct me if I am wrong, but you would like to get 

```js
{
  locationType: 'city',
  data: [ { // only one element
    mvFdwLocationParentCode: 'tr-63',
    mvFdwLocationParentCode: 'POINT(...)',
    // ... and more properties
  } ]
}```

A [collection operator](https://www.mongodb.com/docs/realm/realm-query-language/#collection-operators) or [subquery](https://www.mongodb.com/docs/realm/realm-query-language/#subqueries) might be useful.

**Comment by MuratAkbaba006** [comment]
yes you are correct, this is what I would like to get but none of those methods gave me correct result.

![Screen Shot 2023-06-05 at 09 50 13](https://github.com/realm/realm-js/assets/44754934/3702b4c4-6fc0-4d35-ae0c-6414a0453d63)
![Screen Shot 2023-06-05 at 09 59 02](https://github.com/realm/realm-js/assets/44754934/974e4ea6-80f1-470e-9df4-0d986cc5845f)
![Screen Shot 2023-06-05 at 10 01 19](https://github.com/realm/realm-js/assets/44754934/8479ca4b-d6ab-4cbb-b562-c38eefaa7f23)
![Screen Shot 2023-06-05 at 10 08 41](https://github.com/realm/realm-js/assets/44754934/594c69e0-8f23-4c1b-aff9-f36e4cc6aa84)

those are what I have tried based on your suggestion. They are giving the correct embedded list but I need to filter inside of embedded list just as you described.

**Comment by MuratAkbaba006** [comment]
Hello, I stopped using embedded list. I inspired by this issues and I handled my problem
https://github.com/realm/realm-js/issues/3142
Thank you for clarification
